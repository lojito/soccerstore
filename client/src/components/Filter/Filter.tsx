import { useState, type FC } from "react";
import { styled } from "styled-components";

const DivWrapper = styled.div`
  text-align: center;
  margin-top: 20px;
`;

const Button = styled.button`
  margin-right: 15px;
  margin-bottom: 10px;
  border: 1px solid black;
  border-radius: 6px;
  padding: 5px;
  opacity: ${({ className }) => (className === "selected" ? 0.5 : 1)};

  &:hover {
    cursor: pointer;
  }
`;

export type Category =
  | "balones"
  | "tacos"
  | "shorts"
  | "camisetas"
  | "espinilleras"
  | "guantes"
  | "medias"
  | "chaquetas"
  | "todos"
  | "favorites";

type Props = {
  isLoggedIn: boolean;
  onFilterCategory: (filterCategory: Category) => void;
};

const Filter: FC<Props> = ({ onFilterCategory, isLoggedIn }) => {
  const [categorySelected, setCategorySelected] = useState<Category>("todos");

  const handleClick = (category: Category) => {
    onFilterCategory(category);
    setCategorySelected(category);
  };

  const categories: Array<Category> = [
    "balones",
    "tacos",
    "shorts",
    "camisetas",
    "espinilleras",
    "guantes",
    "medias",
    "chaquetas",
    "todos",
  ];

  if (isLoggedIn) {
    categories.push("favorites");
  }

  const renderCategories = categories.map((category) => {
    return (
      <Button
        key={category}
        className={categorySelected === category ? "selected" : ""}
        type="button"
        onClick={() => {
          handleClick(category);
        }}
      >
        {category[0].toUpperCase() + category.substring(1)}
      </Button>
    );
  });

  return <DivWrapper>{renderCategories}</DivWrapper>;
};

export default Filter;
