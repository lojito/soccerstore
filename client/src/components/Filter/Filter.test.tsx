import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Filter from "./Filter";

describe("Logo", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <Filter isLoggedIn onFilterCategory={vi.fn} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should not render the Favorite button", () => {
    render(<Filter isLoggedIn={false} onFilterCategory={vi.fn} />);

    const button = screen.queryByRole("button", { name: "Favorites" });

    expect(button).not.toBeInTheDocument();
  });

  it("should call the onFilterCategory prop", async () => {
    const user = userEvent.setup();
    const onFilterCategory = vi.fn();
    render(<Filter isLoggedIn onFilterCategory={onFilterCategory} />);

    await user.click(screen.getByRole("button", { name: "Favorites" }));

    expect(onFilterCategory).toHaveBeenCalledTimes(1);
  });
});
