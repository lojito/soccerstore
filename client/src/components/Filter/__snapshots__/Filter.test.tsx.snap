// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh cMiZIK"
  >
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Balones
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Tacos
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Shorts
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Camisetas
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Espinilleras
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Guantes
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Medias
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Chaquetas
    </button>
    <button
      class="sc-guDLey gqjkqk selected"
      type="button"
    >
      Todos
    </button>
    <button
      class="sc-guDLey jbovde"
      type="button"
    >
      Favorites
    </button>
  </div>
</DocumentFragment>
`;
